tests/simple/main.cpp

```diff
@@ -510,6 +510,67 @@ int main() {
                             .SetCompressionMethod(CompressionMethod::LZ4));
             RunTests(client);
         }
+
+        {
+            ClientOptions::HostPort correct_host_port = ClientOptions::HostPort("localhost", 9000);
+            Client client(ClientOptions()
+                              .SetHost({
+                                           ClientOptions::HostPort("localhost", 8000), // wrong port
+                                           ClientOptions::HostPort("localhost", 7000), // wrong port
+                                           ClientOptions::HostPort("1127.91.2.1"), // wrong host
+                                           ClientOptions::HostPort("1127.91.2.2"), // wrong host
+                                           ClientOptions::HostPort("notlocalwronghost"), // wrong host
+                                           ClientOptions::HostPort("another_notlocalwronghost"), // wrong host
+                                           correct_host_port,
+                                           ClientOptions::HostPort("localhost", 9001), // wrong port
+                                           ClientOptions::HostPort("1127.911.2.2"), // wrong host
+                                       })
+                              .SetPingBeforeQuery(true));
```

Comment:
BTW, [this](https://github.com/ClickHouse/clickhouse-cpp/pull/139#discussion_r790474521) allows simplifying client code:
```suggestion
            ClientOptions::Endpoint correct_endpoint{"localhost", 9000};
            Client client(ClientOptions()
                              .SetEndpoints({
                                           {"localhost", 8000}, // wrong port
                                           {"localhost", 7000}, // wrong port
                                           {"1127.91.2.1"}, // wrong host
                                           {"1127.91.2.2"}, // wrong host
                                           {"notlocalwronghost"}, // wrong host
                                           {"another_notlocalwronghost"}, // wrong host
                                           {correct_endpoint},
                                           {"localhost", 9001}, // wrong port
                                           {"1127.911.2.2"}, // wrong host
                                       })
                              .SetPingBeforeQuery(true));
```